LibraBFT/Yasm/Base.agda

```diff
@@ -48,9 +48,11 @@ module LibraBFT.Yasm.Base (ℓ-PeerState : Level) where
       -- Represents an uninitialised PeerState, about which we know nothing whatsoever
       initPS    : PeerState
 
-      -- Initializes a potentially-empty state with an EpochConfig
-      init : PeerId → Genesis → PeerState × List (Action Msg)
+      -- Bootstraps a peer.
+      -- TODO-1 : replace Maybe with Either after adding error type (to mkSysTypeParms ?).
+      bootstrap : PeerId → Genesis → Maybe (PeerState × List (Action Msg))
 
       -- Handles a message on a previously initialized peer.
-      handle : PeerId → Msg → PeerState → PeerState × List (Action Msg)
+      -- TODO-1 : replace Maybe with Either after adding error type (to mkSysTypeParms ?).
```

Comment:
I don't think it makes sense to make `handle` return a `Maybe`.  In case a peer handler encounters an error, it can decide to update the peer state or not (note that it already has one), and it can log errors via the `List (Action Msg)`.  We need a `Maybe` for bootstrapping above, because the `bootstrap` function might be unable to create a peer state at all, e.g., if `Genesis` information is invalid.